src/Features/Core/Portable/ValueTracking/ValueTracker.FindReferencesProgress.cs

```diff
@@ -172,22 +172,43 @@ private async Task TrackMethodInvocationArgumentsAsync(ReferenceLocation referen
             var originalNode = syntaxRoot.FindNode(span);
 
             if (originalNode is null)
-            {
                 return;
-            }
 
             var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
             var invocationSyntax = originalNode.FirstAncestorOrSelf<SyntaxNode>(syntaxFacts.IsInvocationExpression);
             if (invocationSyntax is null)
-            {
                 return;
-            }
 
             var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             var operation = semanticModel.GetOperation(invocationSyntax, cancellationToken);
             if (operation is not IInvocationOperation)
-            {
                 return;
+
+            if (_parameterSymbol is not null)
+            {
+                // Filter out invocations not containing an argument matching _parameterSymbol
+                var containsMatchingArg = false;
+                var invocationDescendants = invocationSyntax.DescendantNodes();
+                foreach (var descendantNode in invocationDescendants)
+                {
+                    if (syntaxFacts.IsArgument(descendantNode))
+                    {
+                        // Ensure this is an argument to the requested invocation
+                        if (invocationSyntax != descendantNode.FirstAncestorOrSelf<SyntaxNode>(syntaxFacts.IsInvocationExpression))
```

Comment:
this isn't a great way to do things.  consider using ISyntaxFacts which will letyou decompose things in a language agnostic fashino, while still being:

1. somewhat strongly typed.
2. ensuring you don't have code that is waslking down infinitely deep and all the way back up again.

Reply:
another approach is to simply use IOp to get the IInvocationOp for your invocation.  Then find the IArg corresponding to your parameter.  And then just see if it is .Implicit or not.

Reply:
Ah, I was making this too complicated. As you mentioned IInvocationOp already has what I need.